public/app/plugins/datasource/loki/components/LokiQueryField.tsx

```diff
@@ -98,6 +92,18 @@ export class LokiQueryField extends React.PureComponent<LokiQueryFieldProps, Lok
     this.setState({ labelsLoaded: true });
   }
 
+  async componentDidUpdate(prevProps: LokiQueryFieldProps) {
+    const {
+      range,
+      datasource: { languageProvider },
+    } = this.props;
+    const refreshLabels = shouldRefreshLabels(range, prevProps.range);
+    // We want to refresh labels when range changes (we round up intervals to a minute)
+    if (refreshLabels) {
+      await languageProvider.fetchLabels();
```

Comment:
nitpick: as this is the last command of the function, i think you can remove the "await" here and remove the "async" at the start of `componentDidUpdate`

in other words, there is no need to wait for the result of `languageProvider.fetchLabels`

Reply:
```suggestion
      languageProvider.fetchLabels();
```